src/Agent.Plugins/PipelineArtifact/FileShareProvider.cs

```diff
@@ -21,47 +24,126 @@ internal class FileShareProvider: IArtifactProvider
         private readonly AgentTaskPluginExecutionContext context;
         private readonly CallbackAppTraceSource tracer;
         private const int defaultParallelCount = 1;
-        private string hostType;
+        private readonly VssConnection connection;
+
         // Default stream buffer size set in the existing file share implementation https://github.com/microsoft/azure-pipelines-agent/blob/ffb3a9b3e2eb5a1f34a0f45d0f2b8639740d37d3/src/Agent.Worker/Release/Artifacts/FileShareArtifact.cs#L154
         private const int DefaultStreamBufferSize = 8192;
 
-        public FileShareProvider(AgentTaskPluginExecutionContext context, CallbackAppTraceSource tracer)
+        public FileShareProvider(AgentTaskPluginExecutionContext context, VssConnection connection, CallbackAppTraceSource tracer)
         {
             this.context = context;
             this.tracer = tracer;
-            this.hostType = context.Variables.GetValueOrDefault("system.hosttype")?.Value;
+            this.connection = connection;
         }
 
-        // For the current existing build artifact task, the artifact name is also created during single download process and we want to preserve this behavior. 
-        public async Task DownloadSingleArtifactAsync(PipelineArtifactDownloadParameters downloadParameters, BuildArtifact buildArtifact, CancellationToken cancellationToken)
+        public async Task DownloadSingleArtifactAsync(PipelineArtifactDownloadParameters downloadParameters, BuildArtifact buildArtifact, CancellationToken cancellationToken) 
```

Comment:
>DownloadSingleArtifactAsync [](start = 26, length = 27)

DownloadSingleArtifactAsync is basically DownloadMultipleArtifactsAsync, it should share code at this point. Just instantiate List<BuildArtifact> and call DownloadMultipleArtifactsAsync directly.